miniapps/gslib/ex1p.cpp

```diff
@@ -0,0 +1,440 @@
+   args.AddOption(&mesh_file_list[2], "-m3", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree) or -1 for"
+                  " isoparametric space.");
+   args.AddOption(&device_config, "-d", "--device",
+                  "Device configuration string, see Device::Configure().");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(&rs_levels[0], "-r1", "--refine-serial",
+                  "Number of times to refine the mesh 1 uniformly in serial.");
+   args.AddOption(&rs_levels[1], "-r2", "--refine-serial",
+                  "Number of times to refine the mesh 2 uniformly in serial.");
+   args.AddOption(&rs_levels[2], "-r3", "--refine-serial",
+                  "Number of times to refine the mesh 3 uniformly in serial.");
+   args.AddOption(&rp_levels[0], "-rp1", "--refine-parallel",
+                  "Number of times to refine the mesh 1 uniformly in parallel.");
+   args.AddOption(&rp_levels[1], "-rp2", "--refine-parallel",
+                  "Number of times to refine the mesh 2 uniformly in parallel.");
+   args.AddOption(&rp_levels[2], "-rp3", "--refine-parallel",
+                  "Number of times to refine the mesh 3 uniformly in parallel.");
+   args.AddOption(&np_list[0], "-np1", "--np1",
+                  "number of MPI ranks for mesh 1");
+   args.AddOption(&np_list[1], "-np2", "--np2",
+                  "number of MPI ranks for mesh 2");
+   args.AddOption(&np_list[2], "-np3", "--np3",
+                  "number of MPI ranks for mesh 3");
+   args.AddOption(&nmeshes, "-nm", "--nm",
+                  "number of meshes");
+   args.AddOption(&rel_tol, "-rt", "--relative tolerance",
+                  "Tolerance for Schwarz iteration convergence criterion.");
+
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   if (myid == 0)
+   {
+      args.PrintOptions(cout);
+   }
+
+   // Setup MPI communicator for each mesh by splitting MPI_COMM_WORLD.
+   MPI_Comm *comml = new MPI_Comm;
+   int color = 0;
+   int npsum = 0;
+   for (int i = 0; i < nmeshes; i++)
+   {
+      npsum += np_list[i];
+      if (myid < npsum) { color = i; break; }
+   }
+
+   MPI_Comm_split(MPI_COMM_WORLD, color, myid, comml);
+   int myidlocal, numproclocal;
+   MPI_Comm_rank(*comml, &myidlocal);
+   MPI_Comm_size(*comml, &numproclocal);
+
+   // Check number of mpi ranks specified for each mesh.
+   MFEM_VERIFY(np_list.Sum() == num_procs, " The individual mpi ranks for each"
+               " of the meshes do not add up to"
+               " the total ranks specified.");
+
+   // Enable hardware devices such as GPUs, and programming models such as
+   // CUDA, OCCA, RAJA and OpenMP based on command line options.
+   Device device(device_config);
```

Comment:
Does this actually work on a device right now? I unfortunately don't have an environment I can easily test this on right now.

Reply:
I don' think any `gslib` code would work on device right now. We are going to try and fix that soon. For now, I have removed the `device` option and I can add it back later.